pipeline/transformer/transformer_tei/src/main/scala/weco/pipeline/transformer/tei/transformers/TeiPhysicalDescription.scala

```diff
@@ -0,0 +1,136 @@
+    *    <height>100mm</height>
+    *    <width>300mm</width>
+    *  </dimensions>
+    * </extent>
+    */
+  private def parseExtent(supportDesc: Node) = {
+    val extent = supportDesc \ "extent"
+    val extentStr = if (extent.exists(_.child.size > 1)) {
+      val extentLabel = extent
+        .flatMap(_.child)
+        .collect { case node if node.label != "dimensions" => node.text.trim }
+        .mkString(" ")
+        .trim
+      val dimensions = parseDimensions(extent)
+      (extentLabel +: dimensions).filterNot(_.isEmpty).mkString("; ")
+    } else extent.text.trim
+    NormaliseText(extentStr)
+  }
+
+  /**
+    * The support contain information about the material, the description and the watermarks present:
+    * <support>Paper, folded in 2.
+    *   <watermark>First watermark very similar to <ref>Mosin and Traljic 6947, 6949, 6956</ref> (saucisson), attested in 1338-50.</watermark>
+    *   <watermark>Second watermark identical with <ref>Mosin and Traljic 5791</ref> (licorne), attested in 1339-44.</watermark>
+    *   <measure type="chainline">Chain distance 43 mm.</measure>
+    *   Added leaves (folios 1 and 198): paper folded in 2.
+    *   <watermark>Watermarks similar to <ref>Piccard 122415</ref> (scissors), attested in 1457.</watermark>
+    *   <measure type="chainline">Chain distance 35mm.</measure>
+    * </support>
+    * We extract the watermarks but we filter out the measure for now
+    */
+  private def parseSupport(supportDesc: Node) = {
+    val support = (supportDesc \ "support")
+    val supportStr = if (support.exists(_.child.size > 1)) {
+      val watermarkStr = (support \ "watermark").text.trim
+      val supportLabel = support
+        .flatMap(_.child)
+        .collect {
+          case node if node.label != "watermark" && node.label != "measure" =>
+            node.text.trim
+        }
+        .mkString(" ")
+        .trim
+      List(
+        supportLabel,
+        if (watermarkStr.nonEmpty) s"Watermarks: $watermarkStr" else "")
+        .filterNot(_.isEmpty)
+        .mkString("; ")
+    } else support.text.trim
+    NormaliseText(supportStr)
+  }
+
+  /**
+    * The dimension block contain information about the unit of measure, width and height.
+    *  <dimensions unit="mm" type="leaf">
+    *    <height>100</height>
+    *    <width>300</width>
+    *  </dimensions>
+    */
+  private def parseDimensions(extent: NodeSeq) =
+    (extent \ "dimensions").map { dimensions =>
+      val height = (dimensions \ "height").text.trim
+      val width = (dimensions \ "width").text.trim
+      val unit = (dimensions \@ "unit").trim
+      val `type` = (dimensions \@ "type").trim
+      val unitStr = if (unit.nonEmpty) unit else ""
+      val heightStr = if (height.nonEmpty) s"height $height".trim else ""
+      val widthStr = if (width.nonEmpty) s"width $width".trim else ""
+      s"${`type`} dimensions: ${List(appendUnit(widthStr, unitStr), appendUnit(heightStr, unitStr))
+        .mkString(", ")}"
```

Comment:
Just one more intermediate variable to break this up, but overall it's really clear:

```suggestion
      val dimensionsStr = List(appendUnit(widthStr, unitStr), appendUnit(heightStr, unitStr)).mkString(", ")

      s"${`type`} dimensions: $dimensionsStr"
```

Will the `type` ever be empty? Do we need one more `.trim`?